File: Plugins/Aspose_Email_Java_for_PHP/src/aspose/email/ProgrammingOutlook/WorkingWithOutlookPersonalStorage/AddMapiTaskToPST.php
<?php
namespace Aspose\Email\ProgrammingOutlook\WorkingWithOutlookPersonalStorage;

use com\aspose\email\MapiTask as MapiTask;
use com\aspose\email\MapiTaskHistory as MapiTaskHistory;
use com\aspose\email\MapiTaskOwnership as MapiTaskOwnership;
use com\aspose\email\PersonalStorage as PersonalStorage;
use com\aspose\email\FileFormatVersion as FileFormatVersion;
use com\aspose\email\StandardIpmFolder as StandardIpmFolder;

use java\util\Date as Date;
class AddMapiTaskToPST{

    public static function run($dataDir=null){

        $task = new MapiTask("To Do", "Just click and type to add new task", new Date(), new Date());
        $task->setPercentComplete(20);
        $task->setEstimatedEffort(2000);
        $task->setActualEffort(20);

        $mapiTaskHistory=new MapiTaskHistory();

        $task->setHistory($mapiTaskHistory->Assigned);
        $task->setLastUpdate(new Date());
        $task->getUsers()->setOwner("Darius");
        $task->getUsers()->setLastAssigner("Harkness");
        $task->getUsers()->setLastDelegate("Harkness");

        $mapiTaskOwnership=new MapiTaskOwnership();
        $task->getUsers()->setOwnership($mapiTaskOwnership->AssignersCopy);

        $personalStorage=new PersonalStorage();
        $fileFormatVersion=new FileFormatVersion();
        $pst = $personalStorage->create($dataDir . "TaskPST.pst", $fileFormatVersion->Unicode);
        $standardIpmFolder=new StandardIpmFolder();
        $task_folder = $pst->createPredefinedFolder("Tasks",$standardIpmFolder->Tasks);
        $task_folder->addMapiMessageItem($task);

        print "Added MapiTask Successfully.".PHP_EOL;
    }
}
?>

File: Plugins/Aspose-Email-Java-for-Jython/asposeemail/ProgrammingOutlook/AddMapiTaskToPST.py
from asposeemail import Settings
from com.aspose.email import MapiTask
from com.aspose.email import MapiTaskHistory
from com.aspose.email import MapiTaskOwnership
from com.aspose.email import PersonalStorage
from com.aspose.email import FileFormatVersion
from com.aspose.email import StandardIpmFolder

from java.util import Date
from java.util import Calendar

class AddMapiTaskToPST:

    def __init__(self):
                        
        dataDir = Settings.dataDir + 'ProgrammingOutlook/WorkingWithOutlookPersonalStorage/AddMapiTaskToPST/'
        
        task = MapiTask("To Do", "Just click and type to add task", Date(), Date())
        task.setPercentComplete(20)
        task.setEstimatedEffort(2000)
        task.setActualEffort(20)

        mapiTaskHistory=MapiTaskHistory()

        task.setHistory(mapiTaskHistory.Assigned)
        task.setLastUpdate(Date())
        task.getUsers().setOwner("Darius")
        task.getUsers().setLastAssigner("Harkness")
        task.getUsers().setLastDelegate("Harkness")

        mapiTaskOwnership=MapiTaskOwnership()
        task.getUsers().setOwnership(mapiTaskOwnership.AssignersCopy)

        personalStorage=PersonalStorage()
        fileFormatVersion=FileFormatVersion
        pst = personalStorage.create(dataDir + "TaskPST.pst", fileFormatVersion.Unicode)
        standardIpmFolder=StandardIpmFolder
        task_folder = pst.createPredefinedFolder("Tasks",standardIpmFolder.Tasks)
        task_folder.addMapiMessageItem(task)

        print "Added MapiTask Successfully.".PHP_EOL
        
 
if __name__ == '__main__':        
    AddMapiTaskToPST()